collect_app/src/main/java/org/odk/collect/android/instancemanagement/LocalInstancesUseCases.kt

```diff
@@ -0,0 +1,104 @@
+package org.odk.collect.android.instancemanagement
+
+import org.odk.collect.android.utilities.FileUtils
+import org.odk.collect.forms.instances.Instance
+import org.odk.collect.forms.instances.InstancesRepository
+import timber.log.Timber
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+object LocalInstancesUseCases {
+    @JvmOverloads
+    @JvmStatic
+    fun createInstanceFileBasedOnFormPath(
+        formDefinitionPath: String,
+        instancesDir: String,
+        clock: () -> Long = { System.currentTimeMillis() }
+    ): File? {
+        val formFileName = formDefinitionPath
+            .substringAfterLast('/')
+            .substringBeforeLast('.')
+
+        return createInstanceFile(
+            formFileName,
+            instancesDir,
+            clock
+        )
+    }
+
+    fun clone(
+        instanceFile: File?,
```

Comment:
I don't think should need to support being passed a null `instanceFile`. Let's deal with that at the "top" level (probably `FormHierarchyViewModel`).